lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -186,19 +198,18 @@ def _populate(self):
                 self.inventory.set_variable(self.node_attrs['ID'], 'ansible_host', self.node_attrs['Status']['Addr'])
                 if self.get_option('include_host_uri', True):
                     self.inventory.set_variable(self.node_attrs['ID'], 'ansible_host_uri',
-                                                "tcp://" + self.node_attrs['Status']['Addr'] + ":2376")
+                                                "tcp://" + self.node_attrs['Status']['Addr'] + ":" + host_uri_port)
                 if self.get_option('verbose_output', True):
                     self.inventory.set_variable(self.node_attrs['ID'], 'docker_swarm_node_attributes', self.node_attrs)
                 if 'ManagerStatus' in self.node_attrs:
                     if self.node_attrs['ManagerStatus'].get('Leader'):
                         # This is workaround of bug in Docker when in some cases the Leader IP is 0.0.0.0
                         # Check moby/moby#35437 for details
+                        swarm_leader_ip = parse_address(self.node_attrs['ManagerStatus']['Addr'])[0] or "0.0.0.0"
                         if self.get_option('include_host_uri', True):
                             self.inventory.set_variable(self.node_attrs['ID'], 'ansible_host_uri', "tcp://" +
```

Comment:
Shouldn't the url begin with `https://` when TLS is enabled? At least that's what `module_utils/docker/common.py` is doing.

Reply:
The `docker.api.client` lib actually recommends `tcp://` as `base_url`. The library does the proper changes so I prefer providing the `tcp://` and let it take care of the rest

Reply:
Good point. TBH, I have no idea why module_utils/docker/common.py does this replacement ­— maybe it once was necessary, but isn't anymore. Or maybe it was never necessary...